lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -95,6 +95,28 @@ bool firrtl::isDuplexValue(Value val) {
       .Default([](auto) { return false; });
 }
 
+/// Return the kind of port this is given the port type from a 'mem' decl.
+static MemOp::PortKind getMemPortKindFromType(FIRRTLType type) {
+  auto portType = type.dyn_cast<BundleType>();
+  if (!portType)
+    return MemOp::PortKind::Debug;
+  // TODO: Avoid set comparisons.
```

Comment:
I expect you can avoid this by checking for `wmode`, then checking for `mask`, then checking for `data` as opposed to looking for all the ports matching.